tmt/steps/discover/__init__.py

```diff
@@ -158,7 +158,7 @@ def load(self) -> None:
             self._tests = {}
 
             for raw_test_datum in raw_test_data:
-                phase_name = raw_test_datum.pop('discover_phase')
+                phase_name = raw_test_datum.pop(key_to_option('discover_phase'))
```

Comment:
Why not just use `discover-phase` directly?

Reply:
I wanted to highlight the "option" (`foo-bar`) is necessary, not the "key" (`foo_bar`). I could have used the string directly, but this way, this particular place is now tracked as a spot where a key is exported requires option-ification. With a mere string, the reason why `-` is in it when the field is called `discover_phase` might be lost, plus when we decide to prepend all exported keys with a `redhat-` prefix, we won't forget about this line.

Reply:
BTW, since we're in the area: how should we call the "option-like" form of a key one can find in fmf files?

There seem to be three words used when speaking about phase data members: fields, keys, and options. For example, the `display-guest` flag of the `html` report plugin:

* the field in `ReportHtmlData` is called `display_guest`. `tmt.utils.container_fields()` will yield the `dataclasses.Field` instance representing this field.

* tmt code also tends to call this particular name "key", as in `ReportHtmlData.keys()` or `ReportHtmlData.values()`.

* then there's the "option" label, used for the `display-guest` name - but there's the `option()` helper which would create `--display-guest` object, which would be a CLI option, and that's where "option" becomes a bit too shady.

Is `display-guest` an option? I guess it's not, `--display-guest` is, `display-guest` is an fmf key for this field. But it's still a user-facing name, and it would be helpful to find a name for this type of label, so I could fix `key_to_option` and `option_to_key` and make their naming correct - they do not operate on CLI options... Well, they do, but they work with fmf keys primarily, and CLI options would become underscored too, but something needs to remove the leading `--`, and so on...

Any idea how to label names written in fmf fields? Beware of "key" though, `keys()` will not yield `display-guest` but `display_guest` which is *not* something one may write in their fmf file...

Or "key", but then we should replace `keys()` with `fields()` and `values()` iteration over keys and their values, but over fields and their values, because, well, `display_guest` would not be key in this found-in-fmf-file sense, `display-guest` would be...

Might look like I have nothing better to do, but this is bothering me: I'd like to make a clear distinction between these names and apply the conversion function at places where they enter tmt code. With some custom types, we should be able to prevent `_` from appearing where `-` should, and vice versa.

Reply:
I see, I thought there might be an intention behind that, thanks for the detailed explanation!

Regarding the naming, here's what comes to my mind:

* **key** like `some-thing` ... is a config input from the `fmf` file (terminology already used in fmf docs), using dashes
* **field** like `some_thing` ... is defined by the `tmt.utils.field` helper, is a python attribute, using underscores
* **option** like `--some-thing` ... is coming from the command line, is defined for `click` with the `--` prefix

How does it sound to you?

Reply:
Sounds good, that's pretty much how it is now, with the added clarification. But, to align the code with this naming scheme, we will need to change some methods, namely `keys()` as that one does not iterate over key names, it iterates over field names, and "keys" is therefore completely misleading. I don't mind doing such a change, it would make it clear in code what's expected and what the code looks at, but it might be a verbose patch :)

Reply:
Ack. And let's document it as well: #2407